build.sc

```diff
@@ -96,32 +97,41 @@ trait CommonModule extends ScalaModule with SbtModule with PublishModule {
   )
 }
 
-class chisel3CrossModule(crossVersionValue: String) extends CommonModule with PublishModule with BuildInfo { m =>
-  // different scala version shares same sources
-  // mill use foo/2.11.12 foo/2.12.12 as millSourcePath by default
-  override def millSourcePath = super.millSourcePath / os.up / os.up
+class chisel3CrossModule(val crossScalaVersion: String) extends CommonModule with BuildInfo {
+  m =>
+  override def mainClass = T {
+    Some("chisel3.stage.ChiselMain")
+  }
 
-  def crossVersion = crossVersionValue
+  override def moduleDeps = super.moduleDeps ++ Seq(macros, core)
 
-  def mainClass = Some("chisel3.stage.ChiselMain")
+  override def scalacPluginClasspath = super.scalacPluginClasspath() ++ Agg(
+    plugin.jar()
+  )
 
-  override def moduleDeps = super.moduleDeps ++ Seq(macros, core) ++ firrtlModule
+  override def scalacOptions = T {
+    super.scalacOptions() ++ Seq(
+      s"-Xplugin:${plugin.jar().path}"
+    )
+  }
```

Comment:
Are both of these necessary? It seems to be like it would be an either/or

Reply:
both of these necessary, I trace back to mill. auto adding `-Xplugin` to compiler didn't contains `scalacPluginClasspath`, I'll try to fix that before long.

Reply:
quite strange: `scalacPluginClasspath` doesn't take any effect, I think this maybe a bug to mill.

Reply:
just found the issue with: `"-verbose", "-Ylog:all"` :(
https://github.com/lihaoyi/mill/blob/bbbf19389ace621c2803b17ddd601623154a084a/scalalib/src/ScalaModule.scala#L19-L24
didn't add `scalacPluginClasspath` to `Tests`. which I think it should be given.

Reply:
just filed a simple fix to mill upstream: https://github.com/lihaoyi/mill/pull/956